gwpy/table/io/losc.py

```diff
@@ -102,17 +101,22 @@ def _mask_column(col):
 @read_with_columns
 @read_with_selection
 def fetch_catalog(catalog, host=DEFAULT_GWOSC_URL):
-    catalog = download_catalog(catalog, host=host)
-    data = catalog["data"]
-    parameters = catalog["parameters"]
+    catalog = fetch_catalog_json(catalog, host=host)
+    data = catalog["events"]
+
+    firstdata = next(iter(data))
+    parameters = list(data[firstdata])
+
+    unitlist = {k: v for k, v in data[firstdata].items() if k.endswith('unit')}
+    parameters = [x for x in parameters if 'unit' not in x]
```

Comment:
We should probably use identically opposite logic to get the non-unit columns as we do in the preceding line to get the unit columns:

```suggestion
    parameters = [x for x in parameters if not x.endswith('unit')]
```